lib/sqlalchemy/dialects/postgresql/hstore.py

```diff
@@ -112,15 +114,29 @@ class MyClass(Base):
         | OperatorClass.CONCATENABLE
     )
 
-    def __init__(self, text_type=None):
+    def __init__(self, text_type=None, use_subscripting_operator=False):
```

Comment:
id rather not have this an opt-in like this, because this becomes a forever parameter that should not be needed, we already know what version of postgresql is in use.   

this should be based on the version of Postgresql that's being used in a way similar to 0f2702caba.   it can use the existing _supports_jsonb_subscripting flag.